rpc/core/types/responses.go

```diff
@@ -254,3 +254,13 @@ type ResultEvent struct {
 	Data   types.TMEventData   `json:"data"`
 	Events map[string][]string `json:"events"`
 }
+
+// Pagination provides pagination information for queries.
+// This allows us to use the same TxSearch API for pruning to return all relevant data,
+// while still limiting public queries to pagination.
+type Pagination struct {
```

Comment:
Shouldn't we add this type to where the `Search()` is declared?

Reply:
Done here https://github.com/cometbft/cometbft/pull/2855/commits/ee94683447e945094726f70a0b73790a0b7a7bb9